openassessment/workflow/models.py

```diff
@@ -364,11 +420,27 @@ def set_score(self, score):
                 'points_possible'.
 
         """
-        sub_api.set_score(
-            self.submission_uuid,
-            score["points_earned"],
-            score["points_possible"]
-        )
+        if not self.staff_score_exists():
+            sub_api.set_score(
+                self.submission_uuid,
+                score["points_earned"],
+                score["points_possible"]
+            )
+
+    def staff_score_exists(self):
+        """
+        Check if a staff score exists for this submission. Cached after first use.
```

Comment:
If it's not working, then I understand. It'd be nice if it could be cached, but if it's too much of a pain to invalidate it, then I'd err on the side of not caching it.